vassal-app/src/main/java/VASSAL/build/module/map/MenuDisplayer.java

```diff
@@ -107,8 +109,15 @@ protected static String getMenuText(KeyCommand keyCommand) {
   // This both eliminates duplicate code AND makes this critical menu-building functionality able to "play well with others".
   // Menu text & behavior can now be custom-classed without needing to override the monster that is MenuDisplayer#createPopup.
   protected static JMenuItem makeMenuItem(KeyCommand keyCommand) {
-
+    // We remember the oldAcceleratorFont so that we can set it back after creating the JMenuItem
+    final Object oldAcceleratorFont = UIManager.get("MenuItem.acceleratorFont");
+    UIManager.put("MenuItem.acceleratorFont", POPUP_MENU_FONT); // This needs to be set prior to creating the JMenuItem
     final JMenuItem item = new JMenuItem(keyCommand.isMenuSeparator() ? MenuSeparator.SEPARATOR_NAME : getMenuText(keyCommand));
+    if (!NamedKeyManager.isNamed(keyCommand.getKeyStroke())) { // If the KeyStroke is named, then there is no accelerator
```

Comment:
@BrentEaston I think you're the person to ask... is this the best way to detect if keystroke is not set?

The older code was using `NamedKeyManager.isNamed(...)` to detect if the keystroke was named, and if so, was not adding the keyboard shortcut to the menu. This is the approach I used here. But it seems weird to me to do this (and rely on the fact that a key that is not named should have a code) rather than explicitly checking directly if the KeyStroke has not been set.

Is there a way to check if the KeyStroke is valid directly?

I looked, expecting to see some kind of sentinel value, like `NamedKeyStroke.NULL_KEYSTROKE`. But I don't see that. Instead I see a keyCode of `57344`. What does this represent? Can it be used as a sentinel value?

Thanks!

Reply:
The relevant code is this:
```
  public static boolean isNamed(KeyStroke k) {
    if (k == null) {
      return false;
    }
    final int code = k.getKeyCode();
    return code >= NAMED_START && code <= NAMED_END;
  }
```
A `NamedKeyStroke` has a code within the range `[NAMED_START,NAMED_END]`.

The only way to check if `KeyStroke` is a _real_ one is to use `NamedKeyManager.isNamed()`, because that's the only thing that knows what `NAMED_START` and `NAMED_END` are.